toolchain/check/type_completion.cpp

```diff
@@ -806,7 +806,51 @@ static auto RequireIdentifiedNamedConstraints(
   return true;
 }
 
-auto RequireIdentifiedFacetType(Context& context,
+// Makes a copy of a Specific from one generic to apply to another given
+// `generic_id`, with a given `Self` argument appended to the end.
+static auto MakeCopyOfSpecificAndAppendSelf(
+    Context& context, SemIR::LocId loc_id, SemIR::SpecificId specific_id,
+    SemIR::GenericId generic_id, SemIR::InstId self_id) -> SemIR::SpecificId {
+  auto source_specific_args_id =
+      context.specifics().GetArgsOrEmpty(specific_id);
+  auto source_specific_args =
+      context.inst_blocks().Get(source_specific_args_id);
+  llvm::SmallVector<SemIR::InstId> arg_ids;
+  arg_ids.reserve(source_specific_args.size() + 1);
+  // Start with the enclosing arguments from the source Specific.
+  llvm::append_range(arg_ids, source_specific_args);
+  // Add the new `Self` argument.
+  arg_ids.push_back(self_id);
+  return MakeSpecific(context, loc_id, generic_id, arg_ids);
+}
+
+static auto GetRequireImplsSpecificSelf(Context& context,
+                                        const SemIR::RequireImpls& require)
```

Comment:
Can you add a comment to this? A simple one would be fine -- it'd just clarify the way things like `require` and `specific` are intended to be parsed (e.g. "require" can be a verb, here it's the noun "`require impls`).